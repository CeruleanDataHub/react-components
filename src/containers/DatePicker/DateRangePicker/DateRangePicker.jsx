import "react-dates/lib/css/_datepicker.css";
import "react-dates/initialize";

import moment from "moment";
import PropTypes from "prop-types";
import React, { useState } from "react";
import { DateRangePicker as DateRangePickerWrapper } from "react-dates";
import momentPropTypes from "react-moment-proptypes";
import styled from "styled-components";

import { Icon } from "../../Icon/Icon";
import { Typography } from "../../Typography/Typography";
import { DateRangePickerStyles } from "../datepickerStyles";

const Container = styled.div`
  ${DateRangePickerStyles};
`;

const CalendarIcon = styled.span`
  font-size: 1.25em;
`;

export const DateRangePicker = ({
  name,
  startDate,
  endDate,
  onDatesChange,
  dateFormat
}) => {
  const [startDateValue, setStartDateValue] = useState(startDate);
  const [endDateValue, setEndDateValue] = useState(endDate);
  const [focusedInputValue, setFocusedInputValue] = useState(null);

  return (
    <Container>
      <Typography fontFamily="openSans">
        <DateRangePickerWrapper
          onDatesChange={dates => {
            setStartDateValue(dates.startDate);
            setEndDateValue(dates.endDate);
            onDatesChange({
              startDate: dates.startDate,
              endDate: dates.endDate
            });
          }}
          startDate={startDateValue}
          startDateId={`${name}-start`}
          endDate={endDateValue}
          endDateId={`${name}-end`}
          focusedInput={focusedInputValue}
          onFocusChange={focusedInput => setFocusedInputValue(focusedInput)}
          hideKeyboardShortcutsPanel
          customInputIcon={<Icon name="calendar" as={CalendarIcon} />}
          displayFormat={dateFormat}
        />
      </Typography>
    </Container>
  );
};

DateRangePicker.propTypes = {
  /** Name for the date picker, used as the component ID */
  name: PropTypes.string.isRequired,
  /** Starting date to be shown on date picker. Defaults to current day. */
  startDate: momentPropTypes.momentObj,
  /** End date to be shown on date picker. Defaults to current day. */
  endDate: momentPropTypes.momentObj,
  /** Event listened for date change */
  onDatesChange: PropTypes.func.isRequired,
  /** How the date should be displayed */
  dateFormat: PropTypes.string
};

const currentMoment = moment();

DateRangePicker.defaultProps = {
  startDate: currentMoment,
  endDate: currentMoment,
  dateFormat: "DD.MM.yyyy"
};
