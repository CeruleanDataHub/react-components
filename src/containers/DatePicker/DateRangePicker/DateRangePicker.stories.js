import { action } from "@storybook/addon-actions";
import moment from "moment";
import React from "react";

import { DateRangePicker } from "./DateRangePicker";

export default {
  title: "DateRangePicker",
  component: DateRangePicker
};

export const DateRangePickerStory = () => (
  <div style={{ minHeight: 400 }}>
    <DateRangePicker name="story" onDatesChange={action("date changed")} />
  </div>
);

export const DateRangePickerCustomRangeStory = () => (
  <div style={{ minHeight: 400 }}>
    <DateRangePicker
      name="story-custom-range"
      onDatesChange={action("date changed")}
      startDate={moment(new Date(2020, 6, 1))}
      endDate={moment(new Date(2020, 6, 12))}
    />
  </div>
);

export const DateRangePickerCustomFormatStory = () => (
  <div style={{ minHeight: 400 }}>
    <DateRangePicker
      name="story"
      onDatesChange={action("date changed")}
      dateFormat="MM/DD/yyyy"
    />
  </div>
);
