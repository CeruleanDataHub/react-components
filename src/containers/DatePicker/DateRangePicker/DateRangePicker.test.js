import { mount } from "enzyme";
import MockDate from "mockdate";
import moment from "moment";
import React from "react";
import renderer from "react-test-renderer";

import { DateRangePicker } from "./DateRangePicker";

const mockDate = new Date("2020-06-19");

const before = () => {
  MockDate.set(mockDate);
};

const after = () => {
  MockDate.reset();
};

describe("DateRangePicker", () => {
  beforeAll(before);
  afterAll(after);

  it("should render", () => {
    const component = renderer.create(
      <DateRangePicker name="test" onDatesChange={() => null} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with custom date format", () => {
    const component = renderer.create(
      <DateRangePicker
        name="test-format"
        onDatesChange={() => null}
        dateFormat="MM/DD/yyyy"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should fire onDatesChange event handler", () => {
    const handleClick = jest.fn();
    const wrapper = mount(
      <DateRangePicker name="test-onDatesChange" onDatesChange={handleClick} />
    );
    expect(handleClick).not.toHaveBeenCalled();
    wrapper.props().onDatesChange({ startDate: moment(), endDate: moment() });
    expect(handleClick).toHaveBeenCalled();
  });

  it("should call onDatesChange event handler when date is changed", () => {
    const handleClick = jest.fn();
    const wrapper = mount(
      <DateRangePicker name="test-onDatesChange" onDatesChange={handleClick} />
    );
    const event = {
      target: {
        value: "2020-07-12"
      }
    };
    expect(handleClick).not.toHaveBeenCalled();
    wrapper
      .find("input")
      .first()
      .simulate("change", event);

    expect(handleClick).toHaveBeenCalled();
  });

  // TODO: this test seems like a stub for now, it's only for improving coverage
  it("should change focus of the inputs", () => {
    const wrapper = mount(
      <DateRangePicker name="test-onclick" onDatesChange={() => null} />
    );
    wrapper
      .find("input")
      .first()
      .simulate("focus");
  });
});

describe("DateRangePicker with given date", () => {
  beforeAll(before);
  afterAll(after);

  it("should render DateRangePicker starting from date provided", () => {
    const component = renderer.create(
      <DateRangePicker
        name="start-end-end"
        onDatesChange={() => null}
        dateFormat="MM/DD/yyyy"
        startDate={moment("2020-07-01", "YYYY-MM-DD")}
        endDate={moment("2020-07-12", "YYYY-MM-DD")}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
