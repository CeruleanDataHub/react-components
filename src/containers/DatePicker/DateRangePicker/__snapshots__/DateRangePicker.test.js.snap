// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker dark should render 1`] = `
<div
  className="sc-AxhCb hplRGN"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-start"
              name="test-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-end"
              name="test-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker dark should render with custom date format 1`] = `
<div
  className="sc-AxhCb hplRGN"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-start"
              name="test-format-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-end"
              name="test-format-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker light should render 1`] = `
<div
  className="sc-AxhCb jzImlU"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-start"
              name="test-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-end"
              name="test-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker light should render with custom date format 1`] = `
<div
  className="sc-AxhCb jzImlU"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-start"
              name="test-format-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-end"
              name="test-format-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker with given date should render DateRangePicker starting from date provided 1`] = `
<div
  className="sc-AxhCb jzImlU"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-start-end-end-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="start-end-end-start"
              name="start-end-end-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-start-end-end-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-start-end-end-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="start-end-end-end"
              name="start-end-end-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-start-end-end-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker with given date should render DateRangePicker starting from date provided 2`] = `
<div
  className="sc-AxhCb hplRGN"
>
  <div
    className="sc-AxiKw ixAZyM"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
        >
          <button
            aria-label="Interact with the calendar and add the check-in date for your trip."
            className="DateRangePickerInput_calendarIcon DateRangePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="sc-AxhUy eNHrrB sc-AxjAm jWlVxK lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-start-end-end-start"
              aria-label="Start Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="start-end-end-start"
              name="start-end-end-start"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-start-end-end-start"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-start-end-end-end"
              aria-label="End Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="start-end-end-end"
              name="start-end-end-end"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="End Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-start-end-end-end"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
